### PR TITLE
Fix & clean up Feast CLI commands

### PR DESCRIPTION
**Pull request by tsotnet**
Signed-off-by: Tsotne Tabidze <tsotne@tecton.ai>

**What this PR does / why we need it**:
* Remove repo_path argument from commands: apply, registry-dump, teardown, materialize, materialize-incremental
* Remove commands: config, feature-tables, projects
* Misc
    * Add help text to `init` command
    * Provide only SDK version in `version` command
    * Add help text including links to doc & Feast Slack to `feast` command
    * Add feature views to `registry-dump` command

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Fix & clean up Feast CLI commands
```

### CODE REVIEW

File: sdk/python/feast/cli.py [hunk 1]
```diff
@@ -246,173 +170,46 @@ def _get_labels_dict(label_str: str) -> Dict[str, str]:
...
-
 @cli.command("apply")
-@click.argument(
-    "repo_path", type=click.Path(dir_okay=True, exists=True), default=Path.cwd
-)
-def apply_total_command(repo_path: str):
+def apply_total_command():
     """
-    Applies a feature repo
+    Apply a feature repo
```
Comment (woop) [inline sdk/python/feast/cli.py:176]: Should we have a more descriptive message here? `Create or update a feature store deployment`?

Comment (tsotnet) [reply sdk/python/feast/cli.py:176]: We have couple of commands that use `X a feature repo` as a help message:
```
  apply                    Apply a feature repo
  init                     Initialize a feature repo
  teardown                 Tear down infra for a feature repo
```

For consistency, how about changing them to
```
  apply                    Create or update a feature store deployment
  init                     Initialize a feature store deployment
  teardown                 Tear down deployed feature store infrastructure 
```
?

Comment (woop) [reply sdk/python/feast/cli.py]: Sure

**Review by woop — APPROVED** [review]

**Pull request merged**